share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
Ok, I think I now as well finally understand what the issue is :): non-standard `pthreadtypes.h` occasionally gets included from "weird" places. I am really not sure what a portable solution would be.
To me it seems that we can either:
1. Use system-wide pthreads type definitions and just perform linking of our models (i.e., C files).
2. Stick more or less with what we have now and try to figure out a reasonable set of "guard macros".
I am somehow a bit worried that option (1) would create problems that we have not even envisioned, in addition to the ones mentioned above. Not sure though.
Which one should we go with? Opinions?
Note that we are also maybe over-analyzing this a bit since our current pthreads benchmarks are rather smallish. If we ever get into checking large pthreads projects, we can revisit this discussion. At that point we would also have a concrete project to go after to drive these decisions.